src/components/LGridLayer.vue

```diff
@@ -0,0 +1,89 @@
+<script>
+import { onMounted, onUnmounted, ref, inject, nextTick, h, render } from "vue";
+import { remapEvents, propsBinder } from "../utils.js";
+import {
+  props as gridLayerProps,
+  setup as gridLayerSetup,
+} from "../functions/gridLayer";
+
+export default {
+  props: {
+    ...gridLayerProps,
+    childRender: {
+      type: Function,
+      required: true,
+    },
+  },
+  setup(props, context) {
+    const leafletRef = ref({});
+    const tileComponents = ref({});
+    const root = ref(null);
+    const ready = ref(false);
+
+    const addLayer = inject("addLayer");
+
+    const { options, methods } = gridLayerSetup(props, leafletRef);
+
+    onMounted(async () => {
+      const { GridLayer, DomEvent } = await import(
+        "leaflet/dist/leaflet-src.esm"
+      );
+
+      methods.onUnload = (e) => {
+        const key = leafletRef.value._tileCoordsToKey(e.coords);
+        if (tileComponents[key]) {
+          tileComponents[key].innerHTML = "";
+          tileComponents[key] = undefined;
```

Comment:
In Vue2Leaflet, the `onUnload` handler actually destroys/removes the created elements, and calls `delete` on the tile component entry. Is there a reason to leave the property key around on the object, but set to `undefined`, instead of actually deleting it? What's the advantage of emptying the `innerHTML` on the object, before removing the reference to it?

Reply:
The reason why I did not use `delete` is because I am generally afraid, and memory wise I think is equivalent to unasign the object.

The reason why I am removing the innerHTML is that is by any chance there is a reference to this dom element somewhere else, we are at least removing the markup.

WDYT?